app/bundles/LeadBundle/Field/Dispatcher/FieldColumnDispatcher.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/*
+ * @copyright   2018 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\LeadBundle\Field\Dispatcher;
+
+use Mautic\LeadBundle\Entity\LeadField;
+use Mautic\LeadBundle\Exception\NoListenerException;
+use Mautic\LeadBundle\Field\Event\AddColumnEvent;
+use Mautic\LeadBundle\Field\Exception\AbortColumnCreateException;
+use Mautic\LeadBundle\Field\Settings\BackgroundSettings;
+use Mautic\LeadBundle\LeadEvents;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+
+class FieldColumnDispatcher
+{
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $dispatcher;
+
+    /**
+     * @var BackgroundSettings
+     */
+    private $backgroundSettings;
+
+    public function __construct(EventDispatcherInterface $dispatcher, BackgroundSettings $backgroundSettings)
+    {
+        $this->dispatcher         = $dispatcher;
+        $this->backgroundSettings = $backgroundSettings;
+    }
+
+    /**
+     * @throws AbortColumnCreateException
+     * @throws NoListenerException
+     */
+    public function dispatchPreAddColumnEvent(LeadField $leadField)
+    {
+        $action = LeadEvents::LEAD_FIELD_PRE_ADD_COLUMN;
+
+        if (!$this->dispatcher->hasListeners($action)) {
+            throw new NoListenerException('There is no Listener for this event');
```

Comment:
In my testing with XDebug, this exception is triggered "There is not listener for this event". Not sure if that's expected behavior.

Reply:
I'm looking at the code and I also cannot find any reason for throwing this exception. It's always caught with no code in the catch part. A `return;` would do the same job. But I think that Petr wanted to be explicit about if the column creation will be made by a plugin subscribed to the event or if it created the column in this method.